pysat/_meta.py

```diff
@@ -389,177 +390,113 @@ def __str__(self, recurse=True):
             # check name of attributes against existing attribute names
             # if attribute name exists somewhere, then case of existing attribute
             # will be enforced upon new data by default for consistency
-            # keys are pulled out below first since dictionary will be modified
-            # only want to iterate over current keys
             keys = [i for i in input_data]
-            for key in keys:
-                new_key = self.attr_case_name(key)
-                input_data[new_key] = input_data.pop(key)
-
-            # time to actually add the metadata
-            if len(name) > 0:
-                # make sure there is still something to add
-                new = DataFrame(input_data, index=name)
-                for item_name, item in new.iterrows():
-                    if item_name not in self:
-                        # this lets data in that could break the system
-                        # when the user tries to modify metadata
-                        # self.data = self.data.append(item)
-                        # thus, instead we take a longer route
-                        # with the for loop below which adds everything one
-                        # by one
-                        new_item_name = item_name
-                    else:
-                        # info already exists, update with new info
-                        new_item_name = self.var_case_name(item_name)
-                    # time to actually add the info
-                    for item_key in item.keys():
-                        # disallow column name children
-                        if item_key not in ['children', 'meta']:
-                            # print ('new_item_name', new_item_name)
-                            # print ('item_key', item_key)
-                            # print ('item[item_key]', item[item_key])
-                            to_be_set = item[item_key]
-                            if hasattr(to_be_set, '__iter__') and not isinstance(to_be_set, basestring):
-                                if isinstance(to_be_set[0], basestring):
-                                    self.data.loc[new_item_name, item_key] = '\n\n'.join(to_be_set)
-                                else:
-                                    warnings.warn(' '.join(('Array elements are disallowed in meta.',
-                                                  'Dropping input :', item_key)))
+            for name in keys:
+                new_name = self.attr_case_name(name)
+                if new_name != name:
+                    input_data[new_name] = input_data.pop(name)
+
+            # time to actually add the metadata           
+            for key in input_data:
+                if key not in ['children', 'meta']:
+                    for i, name in enumerate(names):
+                        to_be_set = input_data[key][i]
+                        if hasattr(to_be_set, '__iter__') and not isinstance(to_be_set, basestring):
```

Comment:
Minor point, but basestring doesn't have \_\_iter\_\_, so if you want to check for an iterable collection, you don't have to explicitly exclude strings.

Reply:
Good to know. One of the iterable checks I've tried before did work on strings and lists/arrays/etc.

Reply:
That is really strange that a test is failing. Is it failing on 2.7? The snippets below are from python2.7
>>> a = 'test'
>>> hasattr(a,'__iter__')
False
>>> hasattr(a,'__iter__')
False
>>> a = u'test'
>>> hasattr(a, '__iter__')
False
>>> a = [1,2,3]
>>> hasattr(a, '__iter__')
True

Reply:
Yep, just checked. str has \_\_iter\_\_ in python3. Oh well, there goes that handy trick.